contracts/strategies/StrategyDAI3poolIron.sol

```diff
@@ -0,0 +1,562 @@
+    }
+
+    function balanceOfy3CRVLent() public view returns (uint256) {
+        (, uint256 tBal, , uint256 rate) = ICErc20(cToken).getAccountSnapshot(address(this));
+        return tBal.mul(rate).div(1e18);
+    }
+
+    function balanceOf() public view returns (uint256) {
+        return balanceOfWant().add(balanceOfy3CRVinWant());
+    }
+
+    function migrate(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        require(Controller(controller).approvedStrategies(want, _strategy), "!stategyAllowed");
+        IERC20(cToken).safeTransfer(_strategy, IERC20(cToken).balanceOf(address(this)));
+        IERC20(y3crv).safeTransfer(_strategy, IERC20(y3crv).balanceOf(address(this)));
+        IERC20(_3crv).safeTransfer(_strategy, IERC20(_3crv).balanceOf(address(this)));
+        IERC20(want).safeTransfer(_strategy, IERC20(want).balanceOf(address(this)));
+    }
+
+    function forceD(uint256 _amount) external isAuthorized {
+        drip();
+        IERC20(want).safeApprove(_3pool, 0);
+        IERC20(want).safeApprove(_3pool, _amount);
+        uint256 v = _amount.mul(1e18).div(ICurveFi(_3pool).get_virtual_price());
+        ICurveFi(_3pool).add_liquidity([_amount, 0, 0], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        if (_amount < tank) tank = tank.sub(_amount);
+        else tank = 0;
+
+        uint256 _bal = IERC20(_3crv).balanceOf(address(this));
+        IERC20(_3crv).safeApprove(y3crv, 0);
+        IERC20(_3crv).safeApprove(y3crv, _bal);
+        yvERC20(y3crv).deposit(_bal);
+
+        _bal = IERC20(y3crv).balanceOf(address(this));
+        IERC20(y3crv).safeApprove(cToken, 0);
+        IERC20(y3crv).safeApprove(cToken, _bal);
+        require(ICErc20(cToken).mint(_bal) == 0, "!forceD");
+        urn = ICErc20(cToken).balanceOfUnderlying(address(this));
+    }
+
+    function forceW(uint256 _amt) external isAuthorized {
+        drip();
+        require(ICErc20(cToken).redeemUnderlying(_amt) == 0, "!forceW");
+        urn = ICErc20(cToken).balanceOfUnderlying(address(this));
+        uint256 _before = IERC20(_3crv).balanceOf(address(this));
+        yvERC20(y3crv).withdraw(_amt);
+        uint256 _after = IERC20(_3crv).balanceOf(address(this));
+        _amt = _after.sub(_before);
+
+        IERC20(_3crv).safeApprove(_3pool, 0);
+        IERC20(_3crv).safeApprove(_3pool, _amt);
+        _before = IERC20(want).balanceOf(address(this));
+        ICurveFi(_3pool).remove_liquidity_one_coin(_amt, 0, _amt.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        _after = IERC20(want).balanceOf(address(this));
+        tank = tank.add(_after.sub(_before));
+    }
+
+    function drip() public isAuthorized {
+        uint256 _p = yvERC20(y3crv).getPricePerFullShare();
+        _p = _p.mul(ICurveFi(_3pool).get_virtual_price()).div(1e18);
+        require(_p >= p, "backward");
+        uint256 _u = ICErc20(cToken).balanceOfUnderlying(address(this));
+        require(_u >= urn, "urn leak");
```

Comment:
is there a work around in case this happens to get funds out of other parts? i saw the the `forceD` so in case of emergency im just wondering if there is a chance that iron bank reverts for some reason and we can't proceed with rest of force withdrawals

Reply:
yes. `pop()` is another emergency protection on iron bank. It withdraws `cyy3crv` back to `y3crv`. And funds can be withdrawn in `y3crv` format as a fallback.